rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -348,8 +348,8 @@ namespace ros {
       }
 
       /* Register a new subscriber */
-      template<typename MsgT>
-      bool subscribe(Subscriber< MsgT> & s){
+      template<typename Subscriber>
+      bool subscribe(Subscriber& s){
         for(int i = 0; i < MAX_SUBSCRIBERS; i++){
           if(subscribers[i] == 0){ // empty slot
             subscribers[i] = (Subscriber_*) &s;
```

Comment:
While you're here anyway, please convert this to a proper C++ `static_cast<Subscriber_*>(&s)`.